en_US/extensions/plugins.md

```diff
@@ -65,12 +95,194 @@ my_emqx_plugin
+`my_emqx_plugin_sup.erl` implements a typical supervisor for the plugin.
+
+`my_emqx_plugin.erl` is the main module of the plugin implementing the plugin's logic. In the skeleton, it implements several demonstrational hooks with simple logging. Any other modules may be added to the plugin.
+
+::: tip Note
+
+The application modules and files may be arbitrarily named with the only requirements:
+
+- The application name must be the same as the plugin name.
+- The application module (`_app`) must be named as `{plugin_name}_app`.
+  :::
+
+#### `priv` Directory
+
+The `priv` directory holds the plugin's configuration files and schema.
+
+##### `config.hocon`
+
+This file contains the plugin's initial configuration in [HOCON format](https://github.com/lightbend/config/blob/master/HOCON.md). You can use `config.hocon.example` for quick reference.
+
+##### `config_schema.avsc`
+
+This file defines the schema for the plugin's configuration in [Avro format](https://avro.apache.org/docs/1.11.1/specification/). When present, EMQX will validate the plugin's configuration against this schema whenever it's updated. The release build will fail if `config.hocon` does not conform to the schema.
+
+Additionally, this file can include UI hints, enabling interactive configuration through the EMQX Dashboard. For reference, see `config_schema.avsc.enterprise.example`.
+
+##### `config_i18n.json`
+
+This file contains translations for the plugin's configuration UI in JSON format. For example:
+
+```
+{
+  "$key": {
+    "zh": "中文翻译",
+    "en": "English translation"
+  },
+  ...
+}
+```
+
+The translations are referenced in the `config_schema.avsc` in UI hints. See `config_i18n.json.example` and `config_schema.avsc.enterprise.example` for more information.
+
+### Package Structure
+
+When a plugin is built into a release, the package structure is as follows:
+
+```
+└── my_emqx_plugin-1.1.0.tar.gz
+    ├── map_sets-1.1.0
+    ├── my_emqx_plugin-0.1.0
+    ├── README.md
+    └── release.json
+```
+
+The tarball includes the compiled applications (as specified in the `relx` section of `rebar.config`), the `README.md` file, and `release.json`, which contains metadata about the plugin.
+
+```
+{
+    "hidden": false,
+    "name": "my_emqx_plugin",
+    "description": "Another amazing EMQX plugin.",
+    "authors": "Anonymous",
+    "builder": {
+        "name": "Anonymous",
+        "contact": "anonymous@example.org",
+        "website": "http://example.com"
+    },
+    "repo": "https://github.com/emqx/emqx-plugin-template",
+    "functionality": "Demo",
+    "compatibility": {
+        "emqx": "~> 5.7"
+    },
+    "git_ref": "unknown",
+    "built_on_otp_release": "27",
+    "emqx_plugrel_vsn": "0.5.1",
+    "git_commit_or_build_date": "2025-04-29",
+    "metadata_vsn": "0.2.0",
+    "rel_apps": [
+        "my_emqx_plugin-0.1.0",
+        "map_sets-1.1.0"
+    ],
+    "rel_vsn": "1.1.0",
+    "with_config_schema": true
+}
+```
+
 ### Test Your Development Environment
```

Comment:
Information from this title ("Test Your Development Environment") up to "Install Plugins via API" is a duplication, we probably should drop it.